stream-chat-android-compose/src/main/java/io/getstream/chat/android/compose/ui/attachments/content/FileAttachmentContent.kt

```diff
@@ -111,50 +127,71 @@ public fun FileAttachmentItem(attachment: Attachment) {
-                    .padding(start = 16.dp, end = 8.dp),
-                horizontalAlignment = Alignment.Start,
-                verticalArrangement = Arrangement.Center
-            ) {
-                Text(
-                    text = attachment.title ?: attachment.name ?: "",
-                    style = ChatTheme.typography.bodyBold,
-                    maxLines = 1,
-                    overflow = TextOverflow.Ellipsis,
-                    color = ChatTheme.colors.textHighEmphasis
-                )
-
-                Text(
-                    text = MediaStringUtil.convertFileSizeByteCount(attachment.fileSize.toLong()),
-                    style = ChatTheme.typography.footnote,
-                    color = ChatTheme.colors.textLowEmphasis
-                )
-            }
-
-            Icon(
-                modifier = Modifier
-                    .align(Alignment.Top)
-                    .padding(end = 2.dp)
-                    .clickable(
-                        interactionSource = remember { MutableInteractionSource() },
-                        indication = rememberRipple(bounded = false)
-                    ) {
-                        ChatClient
-                            .instance()
-                            .downloadAttachment(context, attachment)
-                            .enqueue()
-                    },
-                painter = painterResource(id = R.drawable.stream_compose_ic_file_download),
-                contentDescription = stringResource(id = R.string.stream_compose_download),
-                tint = ChatTheme.colors.textHighEmphasis
-            )
+            FileAttachmentDescription(attachment = attachment)
+            DownloadIcon(attachment = attachment)
         }
     }
 }
 
+/**
+ *  Displays information about the attachment such as
+ *  the attachment title and its size in bytes.
+ *
+ *  @param attachment The attachment for which the information is displayed.
+ */
+@Composable
+private fun FileAttachmentDescription(attachment: Attachment) {
+    Column(
+        modifier = Modifier
+            .fillMaxWidth(0.85f)
+            .padding(start = 16.dp, end = 8.dp),
+        horizontalAlignment = Alignment.Start,
+        verticalArrangement = Arrangement.Center
+    ) {
+        Text(
+            text = attachment.title ?: attachment.name ?: "",
+            style = ChatTheme.typography.bodyBold,
+            maxLines = 1,
+            overflow = TextOverflow.Ellipsis,
+            color = ChatTheme.colors.textHighEmphasis
+        )
+
+        Text(
+            text = MediaStringUtil.convertFileSizeByteCount(attachment.fileSize.toLong()),
+            style = ChatTheme.typography.footnote,
+            color = ChatTheme.colors.textLowEmphasis
+        )
+    }
+}
+
+/**
+ * Downloads the given attachment when clicked.
+ *
+ * @param attachment The attachment to download.
+ */
+@Composable
+private fun RowScope.DownloadIcon(attachment: Attachment) {
```

Comment:
I would maybe rename this to FileDownloadIcon or something, as we might have more `DownloadIcon`s in the future.